roles/ec2-app/tasks/nfs-templates.yml

```diff
@@ -1,43 +1,68 @@
 - block:
+  - set_fact:
+      nfs_host: "{{ groups['nfs-' + target][0] }}"
+  - set_fact:
+      nfs_launch_time: "{{ hostvars[nfs_host]['create_time'][0:19]|regex_replace('T|Z', ' ') }}"
+
+  - block:
```

Comment:
all the `set_fact` actions should be in a single `block`, with the the `when` option linked to each action rather than the block